pyuvsim/simsetup.py

```diff
@@ -964,6 +964,11 @@ def initialize_uvdata_from_params(obs_params):
     time_dict = param_dict['time']
     uvparam_dict.update(parse_time_params(time_dict))
 
+    # There does not seem to be any way to get polarization_array into uvparam_dict, so
+    # let's add it explicitly.
+    if "polarization_array" in param_dict:
+        uvparam_dict['polarization_array'] = param_dict['polarization_array']
+
     # Parse polarizations
     if 'polarization_array' not in uvparam_dict:
         uvparam_dict['polarization_array'] = np.array([-5, -6, -7, -8])
```

Comment:
Somehow, this default is not being set.

Reply:
I've fixed this particular problem by checking if the "polarization_array" is set to None (not just if it doesn't exist), since the `_from_keywords` function explicitly passes it (sometimes as `None`). 

I do however find it strange that the code was previously checking for `"polarization_array"` in `uvparam_dict` but there was absolutely no way to get it _into_ `uvparam_dict`: passing it in `obs_params` does not result in it being in `uvparam_dict`. I don't know if there was a plan to do that which never got implemented, or something else. Thus I don't know if the way I've done it is in alignment with the ideas of the group.

Reply:
Ah okay, that explains it.

The polarization_array has only recently become a settable parameter, so I'm not surprised. It was technically settable through the 'kwargs', but it wasn't documented and an explicit option is better.

Reply:
Oh, should mention its fixed in b88770f040a5c4771dd1faa6ed00f8293a3e8a7c